Source/JavaScriptCore/runtime/ArrayBufferView.cpp

```diff
@@ -41,10 +44,41 @@ ArrayBufferView::ArrayBufferView(
         m_baseAddress = BaseAddress(static_cast<char*>(m_buffer->data()) + m_byteOffset, byteLength);
 }
 
-ArrayBufferView::~ArrayBufferView()
+template<typename Visitor> constexpr decltype(auto) ArrayBufferView::visitDerived(Visitor&& visitor)
 {
-    if (!m_isDetachable)
-        m_buffer->unpin();
+    switch (m_type) {
+    case TypedArrayType::NotTypedArray:
+    case TypedArrayType::TypeDataView:
+        return std::invoke(std::forward<Visitor>(visitor), *static_cast<DataView*>(this));
```

Comment:
Minor stylistic thing, I would write `static_cast<DataView&>(*this)` instead.

Reply:
Thanks, changed.